### DISCUSSION TITLE
Laravel Cashier to override CurlClient with Laravel Http Client

### DISCUSSION THREAD

**Discussion by dawidholka** [laravel/framework#45102] [Ideas]
Hi,

I am just testing the integration with Stripe, when testing I do not want to use Stripe API directly, just create fake responses.

I used for this 
```php
Stripe/ApiRequestor::setHttpClient($fakeClient);
```

I came up with the idea that Laravel Cashier could by default replace the Http Client used by Stripe with the built-in Http Client in Laravel.

Currently Stripe uses [Stripe\HttpClient\CurlClient.php](https://github.com/stripe/stripe-php/blob/master/lib/HttpClient/CurlClient.php)

This would show all requests in Laravel Telescope and make it easier to create fake responses.

What do you think?

**Comment by shahzaibtariqbutt** [comment]
Or we could have option to change the default client, as it could make the testing alot easier?
